java/code/src/com/suse/manager/reactor/messaging/RegisterMinionEventMessageAction.java

```diff
@@ -408,45 +417,38 @@ private void updateKickStartSession(Optional<ActivationKey> activationKey) {
      * @param saltSSHProxyId optional proxy id for saltssh in case it is used
      * @param activationKeyOverride optional label of activation key to be applied to the system
      * @param isSaltSSH true if a salt-ssh system is bootstrapped
+     * @param saltbootInitrd true if saltboot initrd grain is set
      */
     public void finalizeMinionRegistration(String minionId,
                                            String machineId,
                                            Optional<Integer> sshPort,
                                            Optional<Long> saltSSHProxyId,
                                            Optional<String> activationKeyOverride,
-                                           boolean isSaltSSH) {
+                                           boolean isSaltSSH,
+                                           boolean saltbootInitrd) {
         Optional<User> creator = MinionPendingRegistrationService.getCreator(minionId);
-        // take care that custom grains are deployed on the minion before we request them
-        saltApi.syncGrains(new MinionList(minionId));
-        ValueMap grains = new ValueMap(saltApi.getGrains(minionId).orElseGet(HashMap::new));
-        MinionServer minion = migrateOrCreateSystem(minionId, isSaltSSH, activationKeyOverride, machineId, grains);
+        Org org = OrgFactory.getSatelliteOrg();
```

Comment:
Why is this here and not later with proxy detection. I can't see it used before the second declaration.

Reply:
Ah, its needed for catch later. But org in exception will be default org not related to the org of the minion. This doesn't seem right. I think org parameter can be removed as minionID has to be unique across all orgs anyway.